### PR TITLE
Fixes #1 Adding windows building compatibility.

### LINKED ISSUE #1: Windows compatibility
Support for compilation of the project under Windows platform.

### PR DESCRIPTION
**Pull request by alex952**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -8,9 +8,11 @@ endif()
 
 if(WIN32)
   set(CMAKE_SHARED_LINKER_FLAGS "/FORCE:UNRESOLVED")
-  link_directories(${CMAKE_SOURCE_DIR}/thirdparty/pcre/win32/lib)
-  include_directories(${CMAKE_SOURCE_DIR}/thirdparty/getopt)
-  include_directories(${CMAKE_SOURCE_DIR}/thirdparty/pcre/win32/include)
+  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
+  set(BUILD_SHARED_LIBS TRUE)
+  link_directories(${CMAKE_SOURCE_DIR}/src/thirdparty/pcre/win32/lib)
```
Comment (colinpattison) [inline CMakeLists.txt:13]: Alex - where do we stand with having pcre/getopt inside a 3rd party dir? Should we not move these to submodules if we can?

C

File: src/network/sbfTcpConnection.c [hunk 2]
```diff
@@ -1,9 +1,9 @@
 #include "sbfTcpConnection.h"
 #include "sbfTcpConnectionPrivate.h"
 
-#include <sys/socket.h>
-#include <netinet/in.h>
-#include <netinet/tcp.h>
+//#include <sys/socket.h>
```
Comment (colinpattison) [inline src/network/sbfTcpConnection.c]: Delete rather than comment out

C

File: src/thirdparty/getopt/getopt.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
+// XGetopt.h  Version 1.2
+//
+// Author:  Hans Dietrich
+//          hdietrich2@hotmail.com
+//
+// This software is released into the public domain.
+// You are free to use it in any way you like.
+//
+// This software is provided "as is" with no expressed
```
Comment (colinpattison) [inline src/thirdparty/getopt/getopt.h:9]: good to include as is then I guess

C

**Review by colinpattison — COMMENTED** [review]
Alex - can you have a look at the few items highlighted. This should also go to Rich for review

C

File: src/network/sbfTcpConnection.c (hunk 2 above)
Comment (alex952) [reply src/network/sbfTcpConnection.c]: Completely missed reviewing these, as they were moved (couldn't see the diff). I will remove these, as they're included in the compat libraries for linux and darwin.

File: CMakeLists.txt (hunk 1 above)
Comment (alex952) [reply CMakeLists.txt:13]: I don't think these are in any repository anywhere. I grabbed both from an older version of SBF, and included them in our current tree. These dependencies will not change at all, so I think having them in our tree, as opposed to a submodule is easier.

**Pull request merged**